### PR TITLE
Handle multiple reporting restrictions for transcriptions

### PR DESCRIPTION
**Pull request by tonymort**
Handle multiple reporting restrictions for transcriptions

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by johnrace0 — COMMENTED** [review]
Some low level comments/questions

File: src/main/java/uk/gov/hmcts/darts/transcriptions/mapper/TranscriptionResponseMapper.java [hunk 1]
```diff
@@ -52,58 +62,107 @@ TranscriptionUrgencyResponse mapToTranscriptionUrgencyResponse(TranscriptionUrge
...
+            transcriptionResponse.setUrgency(transcriptionEntity.getTranscriptionUrgency().getDescription());
+        }
+        transcriptionResponse.setRequestType(transcriptionEntity.getTranscriptionType().getDescription());
+        transcriptionResponse.setTranscriptionStartTs(transcriptionEntity.getStartTime());
+        transcriptionResponse.setTranscriptionEndTs(transcriptionEntity.getEndTime());
+        transcriptionResponse.setIsManual(transcriptionEntity.getIsManualTranscription());
+
+        mapReportingRestrictions(courtCase, transcriptionResponse);
+
+        // To be removed when FE updated
```
[the remark this answers is not in the document]
Comment (tonymort) [reply src/main/java/uk/gov/hmcts/darts/transcriptions/mapper/TranscriptionResponseMapper.java:115]: front end

File: src/integrationTest/java/uk/gov/hmcts/darts/transcriptions/service/TranscriptionReportingRestrictionsMapperTest.java [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+    void mapsTranscriptionsCorrectlyWhenZeroReportingRestrictionsAssociatedWithCase() {
+        var transcriptionEntity = dartsDatabase.saveWithType(minimalTranscription());
+
+        var transcriptionResponse = transcriptionService.getTranscription(transcriptionEntity.getId());
+
+        assertThat(transcriptionResponse.getReportingRestrictions()).size().isEqualTo(0);
+    }
+
+    @Test
+    void mapsOneReportingRestrictionsCorrectly() {
```
[the remark this answers is not in the document]
Comment (tonymort) [reply src/integrationTest/java/uk/gov/hmcts/darts/transcriptions/service/TranscriptionReportingRestrictionsMapperTest.java:43]: Not been finalised as far as i'm aware

File: src/main/resources/openapi/cases.yaml [hunk 3]
```diff
@@ -647,25 +644,25 @@ components:
...
-          type: string
-          example: "Section 4(2) of the Contempt of Court Act 1981"
-        event_text:
-          type: string
-          example: "event text"
-        event_ts:
-          type: string
-          format: date-time
-          example: "2023-08-21T09:00:00Z"
+#    reporting_restriction:
```
[the remark this answers is not in the document]
Comment (tonymort) [reply src/main/resources/openapi/cases.yaml:647]: yep missed that, moved to common

File: src/main/java/uk/gov/hmcts/darts/common/repository/TranscriptionRepository.java [hunk 4]
```diff
@@ -12,6 +12,7 @@
 
 @Repository
 public interface TranscriptionRepository extends JpaRepository<TranscriptionEntity, Integer> {
+
```
[the remark this answers is not in the document]
Comment (tonymort) [reply src/main/java/uk/gov/hmcts/darts/common/repository/TranscriptionRepository.java:15]: don' recall tbh

File: src/integrationTest/java/uk/gov/hmcts/darts/transcriptions/controller/TranscriptionControllerGetTranscriptionTest.java [hunk 5]
```diff
@@ -115,13 +124,4 @@ private void addCommentToWorkflow(TranscriptionWorkflowEntity workflowEntity, St
...
     }
-
-    @Test
-    void getTranscriptionNotFound() throws Exception {
-        MockHttpServletRequestBuilder requestBuilder = get(ENDPOINT_URL_TRANSCRIPTION, -999);
-        MvcResult response = mockMvc.perform(requestBuilder).andExpect(status().isNotFound()).andReturn();
-        String actualResponse = response.getResponse().getContentAsString();
-        String expectedResponse = getContentsFromFile("tests/transcriptions/transcription/expectedResponseNotFound.json");
-        JSONAssert.assertEquals(expectedResponse, actualResponse, JSONCompareMode.NON_EXTENSIBLE);
-    }
```
[the remark this answers is not in the document]
Comment (tonymort) [reply src/integrationTest/java/uk/gov/hmcts/darts/transcriptions/controller/TranscriptionControllerGetTranscriptionTest.java:126]: yes

**Review by johnrace0 — APPROVED** [review]

**Pull request merged**